WordPress/Classes/Services/SiteAddressService.swift

```diff
@@ -0,0 +1,19 @@
+typealias SiteAddressServiceCompletion = (Result<[SiteAddress]>) -> Void
+
+protocol SiteAddressService {
+    func addresses(for: Locale, completion: @escaping SiteAddressServiceCompletion)
+}
+
```

Comment:
Will you be adapting [this method](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/develop/WordPressKit/DomainsServiceRemote.swift#L137-L162) to conform to this service in a subsequent PR?

Reply:
I  was looking at the UI side of the problem only. We can always adapt both sides of the interface one to another later on